solc/main.cpp

```diff
@@ -53,24 +52,48 @@ static void setDefaultOrCLocale()
 
 int main(int argc, char** argv)
 {
-	setDefaultOrCLocale();
-	solidity::frontend::CommandLineInterface cli(cin, cout, cerr);
-	if (!cli.parseArguments(argc, argv))
+	try
+	{
+		setDefaultOrCLocale();
+		solidity::frontend::CommandLineInterface cli(cin, cout, cerr);
+		if (!cli.parseArguments(argc, argv) || !cli.readInputFiles() || !cli.processInput() || !cli.actOnInput())
+			return 1;
+
+		return 0;
+	}
+	catch (boost::exception const& _exception)
+	{
+		cerr << "Uncaught exception" << boost::diagnostic_information(_exception) << endl;
```

Comment:
There should be some whitespace between the messages:
```suggestion
		cerr << "Uncaught exception: " << boost::diagnostic_information(_exception) << endl;
```
Please run the compiler locally and test that all of these messages are actually printed correctly.